swagger.yaml

```
@@ -4184,6 +4184,10 @@ definitions:
       network_configuration:
         $ref: '#/definitions/network_configuration'
         x-nullable: true
+      platform:
+        type: object
+        $ref: '#/definitions/platform'
+        x-nullable: true
```

If you do this, can you remove the default logic in the code?

```suggestion
        default: 'baremetal'
```

---

What about kube-api which calls internally?

---

Good point.  @filanov what's our convention?

---

As far as i remember default doesn't really work. fist need to check that.
In kube-api we use internal functions so the default (if it works) is done by the infra wrappers that we don't call. 
So if we use this feature we will have to handle it in kube-api. 
There are two options There:
* If it's part of the API (AgentClusterInstall) then we will can add default with webhooks (wip).
* If it's not part of the API we will need to handle it in the code of the controller when calling create.

---

I think that instead of dealing with it twice we can set it on the registration operation (All the method has default values)

---

I agree, the way it's implemented here looks best.

---

+1